src/metadata.rs

```diff
@@ -74,23 +121,26 @@ pub trait Metadata: MajorObject {
 
             if c_res.is_null() {
                 return None;
-            }
-
-            if !c_res.is_null() {
-                for i in 0.. {
-                    let p = unsafe { *c_res.offset(i) };
-                    if p.is_null() {
-                        break;
-                    }
-
-                    metadata.push(_string(p));
-                }
+            } else {
+                metadata.append(&mut _string_array(c_res));
             }
         }
 
         Some(metadata)
     }
 
+    /// Get a single metadata entry, as indicated by `key` and `domain`.
```

Comment:
The other tests are `no_run`, but this one isn't. I'm not sure, but I think we _should_ run them if possible?

Reply:
@lnicola I've been marking all doctests as `no_run` based on prior feedback. Any thoughts on what should be run and what shouldn't? (In this case, it was an oversight on my part).

Reply:
That's because docs.rs doesn't have GDAL, right? Then I suppose `no_run` makes sense.

Reply:
aaaah..... 


Just checked out [this](https://github.com/rust-lang/docs.rs/blob/master/README.md#build-subcommand) test against the `docs.rs` environment without actually publishing to docs.rs... but looks like you have to literally build your own docs.rs and then your docs in a container.

Reply:
You can make them `no_run`, see https://github.com/georust/gdal/blob/master/gdal-sys/build.rs#L89.